crates/neon-runtime/src/napi/bindings.rs

```diff
@@ -0,0 +1,119 @@
+use std::mem::MaybeUninit;
+use lazy_static::lazy_static;
+use libloading::{Library, Symbol};
+
+/* Later we should do:
+#[repr(C)]
+struct NapiEnvStruct {}
+
+#[repr(C)]
+struct NapiValueStruct {}
+
+pub(crate) type NapiEnv = *mut NapiEnvStruct;
+pub(crate) type NapiValue = *mut NapiValueStruct;
+
+But in this sample we still rely on nodejs_sys's types
+*/
+
+pub(crate) type NapiEnv = nodejs_sys::napi_env;
+pub(crate) type NapiValue = nodejs_sys::napi_value;
+
+#[repr(C)]
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+#[allow(non_camel_case_types)]
+#[allow(dead_code)]
+pub(crate) enum NapiStatus {
+    napi_ok,
+    napi_invalid_arg,
+    napi_object_expected,
+    napi_string_expected,
+    napi_name_expected,
+    napi_function_expected,
+    napi_number_expected,
+    napi_boolean_expected,
+    napi_array_expected,
+    napi_generic_failure,
+    napi_pending_exception,
+    napi_cancelled,
+    napi_escape_called_twice,
+    napi_handle_scope_mismatch,
+    napi_callback_scope_mismatch,
+    napi_queue_full,
+    napi_closing,
+    napi_bigint_expected,
+    napi_date_expected,
+    napi_arraybuffer_expected,
+    napi_detachable_arraybuffer_expected,
+    napi_would_deadlock,
+}
+
+pub(crate) struct Napi<'a> {
+    pub napi_get_undefined: Symbol<'a, unsafe extern "C" fn(env: NapiEnv, out: *mut NapiValue) -> NapiStatus>,
```

Comment:
I think the actual declaration is quite hard to read here, because of the `Symbol<>` wrapper and because it's in the middle of the line instead of at the start. Could we make it simpler somehow?